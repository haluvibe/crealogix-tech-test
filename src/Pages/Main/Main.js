import React from "react";
import { Link } from "react-router-dom";

import CounterProvider from "../../Provider/CounterProvider";
import Counter from "../../Components/Counter/Counter";

const Main = () => (
  <>
    <CounterProvider
      render={({ count, handleIncrementCounter, handleDecrementCounter }) => (
        <Counter
          count={count}
          handleIncrementCounter={handleIncrementCounter}
          handleDecrementCounter={handleDecrementCounter}
        />
      )}
    />
    <Link to="/history">History</Link>
  </>
);

export default Main;
