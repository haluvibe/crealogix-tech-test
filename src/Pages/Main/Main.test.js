import React from "react";
import { shallow } from "enzyme";
import Main from "./Main";

describe("Main", () => {
  it("should render correctly with no props", () => {
    const wrapper = shallow(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
