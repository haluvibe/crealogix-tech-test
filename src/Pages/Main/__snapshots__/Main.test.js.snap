// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly with no props 1`] = `
<Fragment>
  <Connect(CounterProvider)
    render={[Function]}
  />
  <Link
    replace={false}
    to="/history"
  >
    History
  </Link>
</Fragment>
`;
